### PR TITLE
links are now validated and working non-relative

### PR DESCRIPTION
**Pull request by adlbrtbsr**

### CODE REVIEW

File: frontend/src/components/newMovieForm.tsx [hunk 1]
```diff
@@ -73,6 +89,21 @@ export const NewMovieForm = () => {
...
+  const handleMovieLink = (url: string) => {
+    if (isUrlValid(url)) {
+      setMovieLink("//" + url);
+    } else {
+      console.log("URL invalid");
+    }
+  };
+
+  function isUrlValid(url: string): boolean {
+    const regexQuery =
```
Comment (LissaGreense) [inline frontend/src/components/newMovieForm.tsx:101]: Move it outside of function, it's long and the variable is reallocated each time the function is called.

Like this:
const URL_REGEX: RegExp = /^(https?:\\/\\/)?((([-a-z0-9]{1,63}\\.)*?[a-z0-9]([-a-z0-9]{0,253}[a-z0-9])?\\.[a-z]{2,63})|((\\d{1,3}\\.){3}\\d{1,3}))(:\\d{1,5})?((\\/|\\?)((%[0-9a-f]{2})|[-\\w\\+\\.\\?\\/@~#&=])*)?$/i;

File: frontend/src/components/newMovieForm.tsx [hunk 2]
```diff
@@ -73,6 +89,21 @@ export const NewMovieForm = () => {
     }
   };
 
+  const handleMovieLink = (url: string) => {
+    if (isUrlValid(url)) {
+      setMovieLink("//" + url);
+    } else {
+      console.log("URL invalid");
```
Comment (LissaGreense) [inline frontend/src/components/newMovieForm.tsx:96]: set is warn or error

File: frontend/src/components/newMovieForm.tsx [hunk 3]
```diff
@@ -15,8 +15,23 @@ export const NewMovieForm = () => {
   const [movieGenre, setMovieGenre] = useState<string>("");
   const [movieCoverLink, setMovieCoverLink] = useState<string>("");
   const [movieDuration, setMovieDuration] = useState<number>(0);
+  const [formIncomplete, setFormIncomplete] = useState<boolean>(true);
   const [posterUrls, setPosterUrls] = useState<string[]>([]);
 
+  useEffect(() => {
+    if (
+      movieTitle == "" ||
```
Comment (LissaGreense) [inline frontend/src/components/newMovieForm.tsx:23]: move to func, pls? ;)

File: frontend/src/components/movieList.tsx [hunk 4]
```diff
@@ -42,7 +42,7 @@ export const MovieList = () => {
           </div>
           <div className="align-content-start movieData">
             <p>
-              <a href={data.link}>link</a>
+              <a href={"//" + data.link}>link</a>
```
Comment (LissaGreense) [inline frontend/src/components/movieList.tsx:45]: what id data.link already contains http or https prefix?

File: frontend/src/components/newMovieForm.tsx (hunk 1 above)
Comment (adlbrtbsr) [reply frontend/src/components/newMovieForm.tsx]: function removed

File: frontend/src/components/newMovieForm.tsx [hunk 5]
```diff
@@ -73,6 +89,21 @@ export const NewMovieForm = () => {
...
+      setMovieLink("//" + url);
+    } else {
+      console.log("URL invalid");
+    }
+  };
+
+  function isUrlValid(url: string): boolean {
+    const regexQuery =
+      "^(https?:\\/\\/)?((([-a-z0-9]{1,63}\\.)*?[a-z0-9]([-a-z0-9]{0,253}[a-z0-9])?\\.[a-z]{2,63})|((\\d{1,3}\\.){3}\\d{1,3}))(:\\d{1,5})?((\\/|\\?)((%[0-9a-f]{2})|[-\\w\\+\\.\\?\\/@~#&=])*)?$";
+    const checkUrl = new RegExp(regexQuery, "i");
```
[the remark this answers is not in the document]
Comment (adlbrtbsr) [reply frontend/src/components/newMovieForm.tsx]: function removed

File: frontend/src/components/newMovieForm.tsx (hunk 2 above)
Comment (adlbrtbsr) [reply frontend/src/components/newMovieForm.tsx]: warn set

File: frontend/src/components/movieList.tsx (hunk 4 above)
Comment (adlbrtbsr) [reply frontend/src/components/movieList.tsx:45]: data will no more contain http

File: frontend/src/components/newMovieForm.tsx (hunk 3 above)
Comment (adlbrtbsr) [reply frontend/src/components/newMovieForm.tsx:26]: ok moved

**Review by LissaGreense — APPROVED** [review]
Great job @adlbrtbsr, next milestone is achieved!

**Pull request merged**